python/src/dataset.rs

```diff
@@ -152,6 +152,34 @@ impl Dataset {
                 .map_err(|err| PyIOError::new_err(err.to_string()))?)
         })
     }
+
+    fn versions(self_: PyRef<'_, Self>) -> PyResult<Vec<PyObject>> {
+        let versions = self_
+            .list_versions()
+            .map_err(|err| PyValueError::new_err(err.to_string()))?;
+        Python::with_gil(|py| {
+            let pyvers: Vec<PyObject> = versions
+                .iter()
+                .map(|v| {
+                    let dict = PyDict::new(py);
+                    dict.set_item("version", v.version).unwrap();
+                    dict.set_item("timestamp", v.timestamp.timestamp()).unwrap();
+                    let tup: Vec<(&String, &String)> = v.metadata.iter().collect();
+                    dict.set_item("metadata", tup.into_py_dict(py)).unwrap();
+                    dict.to_object(py)
+                })
+                .collect::<Vec<_>>()
+                .into_iter()
+                .collect();
+            Ok(pyvers)
+        })
+    }
+}
+
+impl Dataset {
```

Comment:
is this a separated function from other implementations?  why do you need to have a separate `impl Dataset` here.

Reply:
so it's not exposed as pymethod